File: lista de exercicios/questão 6/questao.py
class Motor:
    def __init__(self, potencia):
        self.potencia = potencia

class Carro:
    def __init__(self, marca, modelo, ano, motor):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.motor = motor

    def exibir_detalhes(self):
        print(f"Marca: {self.marca}, Modelo: {self.modelo}, Ano: {self.ano}, Potência do motor: {self.motor.potencia} cavalos")

motor = Motor(200)
carro = Carro("MarcaA", "ModeloX", 2020, motor)
carro.exibir_detalhes()

File: lista de exercicios/questão 6/questao.go
package main

import "fmt"

type Motor struct {
    Potencia int
}

type Carro struct {
    Marca  string
    Modelo string
    Ano    int
    Motor  Motor
}

func (c Carro) ExibirDetalhes() {
    fmt.Printf("Marca: %s, Modelo: %s, Ano: %d, Potência do motor: %d cavalos\n", c.Marca, c.Modelo, c.Ano, c.Motor.Potencia)
}

func main() {
    motor := Motor{Potencia: 200}
    carro := Carro{Marca: "MarcaA", Modelo: "ModeloX", Ano: 2020, Motor: motor}
    carro.ExibirDetalhes()
}
